game/views.py

```diff
@@ -123,8 +133,63 @@ def get(request, game_id):
             'tweets': tweets.request_return_handle(game['name'], DetailPageView.TWEETS_LIMIT),
         }
 
+        if request.user.is_authenticated:
+            context.update({
+                'is_added': game_id in list(request.user.favourites.values_list('game__game_id', flat=True))
+            })
+
         return render(request, 'game/game_detail_page.html', context)
 
+    def post(self, request, game_id):
+        """
+        Adds game to favourites on must button click, deletes game from favourites on unmust button click.
+        Works on both main and musts pages must buttons.
+        """
+
+        if game_id not in list(request.user.favourites.values_list('game__game_id', flat=True)):
+            current_game, created = Game.objects.get_or_create(game_id=game_id)
+            request.user.favourites.create(game=current_game, user=request.user)
+        else:
+            request.user.favourites.get(game__game_id=game_id).delete()
+
+        return HttpResponse(status=200)
+
+
+class MustsPageView(View):
+    """
+    View for musts page: displays favourite user games with the opportunity to soft delete items.
+    Soft delete and add performed by button unmust/must click.
+    """
+
+    GAMES_LIMIT = 18
+    GAMES_PER_PAGE = 12
+
+    @staticmethod
+    def get(request):
+        """Gets favourite user games (soft deleted items are deleted completely) """
```

Comment:
yeah, no need to delete them. They should be still in database, if entry already created then it's needed tick/untick is_deleted flag

Reply:
Have fixed soft delete functionality on all pages, soft deleted items are stored in the database now👍